### PR TITLE
presentation votes count report

### PR DESCRIPTION
**Pull request by romanetar**
ref https://tipit.avaza.com/project/view#!tab=task-pane&groupby=MyTaskDueDate&view=vertical&task=2833271&fileview=grid

Signed-off-by: romanetar <roman_ag@hotmail.com>

### CODE REVIEW

File: app/ModelSerializers/SerializerRegistry.php [hunk 1]
```diff
@@ -229,8 +229,9 @@ private function __construct()
         $this->registry['Presentation'] =
         [
             self::SerializerType_Public => PresentationSerializer::class,
-            self::SerializerType_Private => AdminPresentationSerializer::class,
+            //self::SerializerType_Private => AdminPresentationSerializer::class,
```
Comment (romanetar) [inline app/ModelSerializers/SerializerRegistry.php:232]: Temporarily commented out until we define a particular selection criteria or decide to unregister this superclass

File: app/Models/Foundation/Summit/Events/Presentations/Presentation.php [hunk 2]
```diff
@@ -1734,11 +1734,48 @@ public function setCustomOrder(int $custom_order): void
...
+        $res = $criteria != null ? $this->attendees_votes->matching($criteria) : $this->attendees_votes;
+        return new ArrayCollection($res->toArray());
+    }
+
+    /**
+     * @return array|String[]
+     */
+    public function getVoters($begin_voting_date = null, $end_voting_date = null): array {
+        return $this->getVotesRange($begin_voting_date, $end_voting_date)->map(function ($attendeeVote) {
+            $voter = $attendeeVote->getVoter();
```
Comment (smarcet) [inline app/Models/Foundation/Summit/Events/Presentations/Presentation.php:1769]: dont do this
pass the voter attendee entity

File: app/ModelSerializers/Summit/Presentation/AdminVoteablePresentationSerializer.php [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+        if (!$presentation instanceof Presentation) return [];
+
+        if (!count($relations)) $relations = $this->getAllowedRelations();
+
+        $values = parent::serialize($expand, $fields, $relations, $params);
+
+        $beginVotingDate = $params['begin_attendee_voting_period_date'] ?? null;
+        $endVotingDate = $params['end_attendee_voting_period_date'] ?? null;
+
+        $values['voters'] = $presentation->getVoters($beginVotingDate, $endVotingDate);
```
Comment (smarcet) [inline app/ModelSerializers/Summit/Presentation/AdminVoteablePresentationSerializer.php:50]: here you should fo a for each and call the proper serializer
like foreach($presentation->getVoters($beginVotingDate, $endVotingDate) as $voter){
$res[] =  SerializerRegistry::getInstance()->getSerializer($voter)->serialize(AbstractSerializer::filterExpandByPrefix($expand, $relation));
}

but please just do it is expand contains the proper expand ( in this case voters)
if not u should return the voters id

File: app/ModelSerializers/SerializerRegistry.php (hunk 1 above)
Comment (smarcet) [reply app/ModelSerializers/SerializerRegistry.php:232]: please create. a new SerializerType_
like
SerializerType_Admin

File: app/Http/Controllers/Apis/Protected/Summit/OAuth2SummitEventsApiController.php [hunk 4]
```diff
@@ -408,17 +406,32 @@ public function getAllVoteablePresentations($summit_id)
...
                 (
                     self::getExpands(),
                     self::getFields(),
                     self::getRelations(),
-                    [
-                        'current_user' => $this->resource_server_context->getCurrentUser(true),
-                        'use_cache' => true,
-                    ],
+                    $params,
                     $this->getSerializerType()
```
Comment (smarcet) [inline app/Http/Controllers/Apis/Protected/Summit/OAuth2SummitEventsApiController.php:435]: we need to defined the logic to get the proper serializer here on getSerializerType based on user rites

File: app/ModelSerializers/SerializerRegistry.php (hunk 1 above)
Comment (romanetar) [reply app/ModelSerializers/SerializerRegistry.php]: new type created as requested: SerializerType_Admin_Voteable

File: app/ModelSerializers/Summit/Presentation/AdminVoteablePresentationSerializer.php (hunk 3 above)
Comment (romanetar) [reply app/ModelSerializers/Summit/Presentation/AdminVoteablePresentationSerializer.php]: done as requested

File: app/Http/Controllers/Apis/Protected/Summit/OAuth2SummitEventsApiController.php (hunk 4 above)
Comment (romanetar) [reply app/Http/Controllers/Apis/Protected/Summit/OAuth2SummitEventsApiController.php]: new serializer type getter was created to cover this scenario

File: app/Models/Foundation/Summit/Events/Presentations/Presentation.php (hunk 2 above)
Comment (romanetar) [reply app/Models/Foundation/Summit/Events/Presentations/Presentation.php]: done as requested

**Comment by romanetar** [comment]
please check out last commit

**Review by smarcet — APPROVED** [review]
looks good to me

**Pull request merged**